www/base/src/app/common/directives/loginbar/loginbar.tpl.jade

```diff
@@ -15,9 +15,8 @@ ul.nav.navbar-nav.navbar-right(ng-show="config.auth.name != 'NoAuth'")
 
     li.dropdown(uib-dropdown, ng-hide="anonymous")
         a.dropdown-toggle(uib-dropdown-toggle)
-            .avatar()
-                img(ng-src="avatar?username={{username}}&email={{email}}")
-            span(ng-if="!email")
+            img.avatar(ng-if="config.avatar_methods.length" ng-src="avatar?username={{username}}&email={{email}}")
+            span(ng-if="!config.avatar_methods.length")
```

Comment:
I don't know if there's a better way to show if there's an avatar or not. This previously was looking at if there was an email, but with #5420 the avatar shows more often.

Reply:
The current method sounds reasonable.